api/src/main/java/com/example/api/model/activity/result/TaskResult.java

```diff
@@ -27,16 +31,19 @@ public abstract class TaskResult {
     public abstract boolean isEvaluated();
     public abstract Activity getActivity();
 
-    public TaskResult(Long id, User user, Double pointsReceived, Long sendDateMillis) {
+    public TaskResult(Long id, User user, Double pointsReceived, Long sendDateMillis, BadgeService badgeService)
+            throws WrongUserTypeException, EntityNotFoundException, MissingAttributeException {
         this.id = id;
         this.user = user;
-        this.setPointsReceived(pointsReceived);
+        this.setPointsReceivedAndCheckBadges(pointsReceived, badgeService);
         this.sendDateMillis = sendDateMillis;
     }
 
-    public void setPointsReceived(Double newPoints) {
+    public void setPointsReceivedAndCheckBadges(Double newPoints, BadgeService badgeService)
+            throws WrongUserTypeException, EntityNotFoundException, MissingAttributeException {
         if (pointsReceived == null) user.changePoints(newPoints);
         else user.changePoints(newPoints - pointsReceived);
         pointsReceived = newPoints;
+        badgeService.checkAllBadges();
```

Comment:
Nie znam trochę pełnego kontekstu, być może to musi być w tym setterze. Ale jeżeli nie musi to ja bym tego tutaj nie wrzucał bo to takie trochę łączenie różnych odpowiedzialności w jednej metodzie. Z resztą sama nazwa tej metody - zrób coś AND zrób coś. Mam nadzieję, że wiesz o co mi chodzi 😅 . Jakby się dało to bym to robił w innej metodzie... czy coś 😆 .

Reply:
W teorii można, ale to będzie coś za coś. Wtedy w każdym miejscu gdzie jest result.setPointsReceived() pod tym bedzie musiały być result.checkBadges(). W sumie z punktu widzenia obiektowego programowania to jest to lepsze podejście, bo tak jak mówisz oddzielamy odpowiedzialności, więc jutro to zmienie.

Reply:
A co do errora to zbudowałeś na nowo dockera? Bo tam dodawałem zdjęcie i Ci to nie zczytało, a kopiowany jest cały folder images do dockera, więc powinno działać. Jeżeli to nie to, to sprawdzę jutro.